runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -729,6 +895,45 @@ TR_VectorAPIExpansion::getDataTypeFromClassNode(TR::Compilation *comp, TR::Node
    return fej9->getClassPrimitiveDataType(clazz);
    }
 
+TR_OpaqueClassBlock *
+TR_VectorAPIExpansion::getArrayClassFromDataType(TR::Compilation *comp, TR::DataType type, bool booleanClass)
+   {
+   TR_J9VMBase *fej9 = comp->fej9();
+   J9JavaVM *vm = fej9->getJ9JITConfig()->javaVM;
+   J9Class *j9class;
+
+
+   switch (type)
+      {
+      case TR::Float:
+         j9class = vm->floatArrayClass;
```

Comment:
FYI: Accessing J9JavaVM directly in the optimizer will break JITServer functionality. Typically a frontend function needs to be created to fetch the appropriate field and that function needs to be overridden at the server frontend. PR https://github.com/eclipse-openj9/openj9/pull/20592 shows an example